src/Framework.Cqrs.Extensions.AspNetCore/AspNetCoreResolverBuilder.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace PetProjects.Framework.Cqrs.Extensions.AspNetCore
+{
+    using Microsoft.Extensions.DependencyInjection;
+    using PetProjects.Framework.Cqrs.Commands;
+    using PetProjects.Framework.Cqrs.DependencyResolver;
+    using PetProjects.Framework.Cqrs.Queries;
+
+    internal class AspNetCoreResolverBuilder : IDependencyResolverBuilder
+    {
+        private readonly IServiceCollection collection;
+
+        public AspNetCoreResolverBuilder(IServiceCollection collection)
+        {
+            this.collection = collection;
+        }
+
+        public void RegisterQueryHandlerAsync<THandler, TQuery, TResponse>()
```

Comment:
Can you give an example of how this would be used?

Because from what I can imagine, and I can be wrong, you will need to declare the query and response two times. One for the query handler and another for TQuery and TResponse.

Reply:
servicesCollection.AddCqrsDependencyResolver(builder => builder.
    RegisterQueryHandlerAsync<GetTransactionByIdQueryHandler, GetTransactionByIdQuery, TransactionByIdReadModel>());

Reply:
public class GetTransactionByIdQueryHandler : QueryHandlerAsync<GetTransactionByIdQuery, TransactionByIdReadModel> {}

public class GetTransactionByIdQuery : IQuery<TransactionByIdReadModel> {}

Reply:
I just noticed that I didn't put these methods on the builder returning the builder itself, they are all returning void which sucks if we want to use fluent like method calls.

Reply:
Fixed in ea71b47c46452dff512e80615b5a26f48d471c3d

Reply:
Nvm already tried it.